core/src/main/java/com/scalar/db/storage/jdbc/JdbcService.java

```diff
@@ -123,18 +122,9 @@ public List<Result> scan(Scan scan, Connection connection)
 
   private SelectQuery buildSelectQuery(Scan scan, TableMetadata tableMetadata) {
     if (scan instanceof ScanAll) {
-      ScanAll scanAll = (ScanAll) scan;
-      if (ScalarDbUtils.isRelational(scan)) {
-        return buildSelectQueryForRelationalScan(scanAll, tableMetadata);
-      } else {
-        return buildSelectQueryForScanAll(scanAll, tableMetadata);
-      }
-    } else {
-      return buildSelectQueryForScan(scan, tableMetadata);
+      return buildSelectQuery((ScanAll) scan, tableMetadata);
```

Comment:
After introducing the cross-partition scan concept, I was convinced that we don't have to distinguish `ScanAll` and `RelationalScan` anymore. This is a kind of refactoring, but I included this since it is related to re-organizing relational scan terminology.